web/server/codechecker_server/api/product_server.py

```diff
@@ -90,26 +89,19 @@ def __get_product(self, session, product):
             self.__server.add_product(product)
             server_product = self.__server.get_product(product.endpoint)
 
-        server_product.connect()
         num_of_runs = 0
         runs_in_progress = set()
         latest_store_to_product = ""
-        if server_product.db_status == shared.ttypes.DBStatus.OK:
-            with DBSession(server_product.session_factory) as run_db_session:
-                run_locks = run_db_session.query(RunLock.name) \
-                    .filter(RunLock.locked_at.isnot(None)) \
-                    .all()
-
-                runs_in_progress = set([run_lock[0] for run_lock in run_locks])
-
-                num_of_runs = run_db_session.query(Run).count()
-                if num_of_runs:
-                    last_updated_run = run_db_session.query(Run) \
-                        .order_by(Run.date.desc()) \
-                        .limit(1) \
-                        .one_or_none()
-
-                    latest_store_to_product = last_updated_run.date
+        try:
+            num_of_runs, runs_in_progress, latest_store_to_product = \
+                server_product.get_details()
+        except Exception:
```

Comment:
What type of exception do we expect here SQLAlchemy? Probably it would be better to handle a more specific exception.

Reply:
I think we should leave it this way because we will get different errors for different sql drivers:
```
ProgrammingError: (pg8000.core.ProgrammingError) (u'FATAL', u'FATAL', u'3D000', u'database "dummy" does not exist', u'postinit.c', u'855', u'InitPostgres', u'', u'')
```
```
(sqlite3.OperationalError) no such table: run_locks
```
```
OperationalError: (psycopg2.OperationalError) FATAL:  database "dummy" does not exist
```